File: bike-shop/bike-shop/backend/src/main/java/com/example/bike_shop/service/ReviewService.java
package com.example.bike_shop.service;

import com.example.bike_shop.dto.ReviewDTO;

import java.util.List;

public interface ReviewService {
    List<ReviewDTO> getByProduct(Long productId, Long currentUserId);
    ReviewDTO create(Long userId, Long productId, ReviewDTO dto);
    void delete(Long userId, Long reviewId, boolean isAdmin);
    boolean canReview(Long userId, Long productId);
    ReviewDTO reply(Long reviewId, String reply);
    ReviewDTO toggleLike(Long userId, Long reviewId);
    List<ReviewDTO> getTestimonials();
    List<ReviewDTO> getAllForAdmin();
}

File: bike-shop/bike-shop/frontend/src/services/reviewService.js
import axiosClient from "./axiosConfig";

const reviewService = {
  getByProduct: (productId) => axiosClient.get(`/reviews/product/${productId}`),
  canReview: (productId) => axiosClient.get(`/reviews/product/${productId}/can-review`),
  create: (productId, data) => axiosClient.post(`/reviews/product/${productId}`, data),
  delete: (reviewId) => axiosClient.delete(`/reviews/${reviewId}`),
  reply: (reviewId, reply) => axiosClient.put(`/reviews/${reviewId}/reply`, { reply }),
  toggleLike: (reviewId) => axiosClient.post(`/reviews/${reviewId}/like`),
  getTestimonials: () => axiosClient.get("/reviews/latest"),
  getAllForAdmin: () => axiosClient.get("/reviews/admin/all"),
};

export default reviewService;
